dotCMS/src/main/java/com/dotcms/saml/SamlConfigurationServiceImpl.java

```diff
@@ -0,0 +1,318 @@
+                break;
+            case DOT_SAML_LOGOUT_PATH_VALUES:
+                defaultParams.setDotSamlLogoutPathValues(value);
+                break;
+            case DOT_SAML_LOGOUT_SERVICE_ENDPOINT_URL:
+                defaultParams.setDotSamlLogoutServiceEndpointUrl(value);
+                break;
+            case DOT_SAML_MESSAGE_LIFE_TIME:
+                try {
+                    defaultParams.setDotSamlMessageLifeTime(Integer.parseInt(value));
+                } catch (Exception ex) {
+                    Logger.warn(DotsamlDefaultPropertiesService.class,
+                            INTEGER_PARSE_ERROR + property.getPropertyName() + ":" + value);
+                }
+                break;
+            case DOT_SAML_REMOVE_ROLES_PREFIX:
+                defaultParams.setDotSamlRemoveRolesPrefix(value);
+                break;
+            case DOT_SAML_ROLES_ATTRIBUTE:
+                defaultParams.setDotSamlRolesAttribute(value);
+                break;
+            case DOT_SAML_SERVICE_PROVIDER_CUSTOM_CREDENTIAL_PROVIDER_CLASSNAME:
+                defaultParams.setDotSamlServiceProviderCustomCredentialProviderClassname(value);
+                break;
+            case DOT_SAML_VERIFY_SIGNATURE_CREDENTIALS:
+                defaultParams.setDotSamlVerifySignatureCredentials(Boolean.parseBoolean(value));
+                break;
+            case DOT_SAML_VERIFY_SIGNATURE_PROFILE:
+                defaultParams.setDotSamlVerifySignatureProfile(Boolean.parseBoolean(value));
+                break;
+            case DOTCMS_SAML_CLEAR_LOCATION_QUERY_PARAMS:
+                defaultParams.setDotcmsSamlClearLocationQueryParams(Boolean.parseBoolean(value));
+                break;
+            case DOTCMS_SAML_LOGIN_UPDATE_EMAIL:
+                defaultParams.setDotcmsSamlLoginEmailUpdate(Boolean.parseBoolean(value));
+                break;
+            case DOT_SAML_ALLOW_USER_SYNCHRONIZATION:
+                defaultParams.setAllowUserSynchronization(Boolean.parseBoolean(value));
+                break;
+            default:
+                Logger.warn(DotsamlDefaultPropertiesService.class,
+                        NOT_FOUND_ERROR + property.getPropertyName() + ":" + value);
+                break;
+        }
+    }
+
+    @Override
+    public String getConfigAsString(final IdentityProviderConfiguration identityProviderConfiguration, final SamlName samlName) {
+
+        try {
+
+            final String value = identityProviderConfiguration.getOptionalProperties().containsKey(samlName.getPropertyName())?
+                    (String) identityProviderConfiguration.getOptionalProperties().get(samlName.getPropertyName()):
+                    this.getDefaultStringParameter(samlName);
+
+            Logger.debug(this,
+                    ()-> "Found " + samlName.getPropertyName() + " : " + ((value == null) ? "null" : value));
+
+            return value;
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)